modules/c.nix

```diff
@@ -0,0 +1,130 @@
+{ pkgs, ... }:
+let
+  clang = pkgs.clang_14;
+  compile = pkgs.writeShellScriptBin "compile" ''
+    CFLAGS="$CFLAGS -g -Wno-everything -pthread -lm"
+    FILE="$1"     # a .c file
+    MODE="$2"     # mode can be:
+                  #   single - compile just one .c file, or
+                  #   all - compile all .c files
+    DEBUG="$3"    # debug can be:
+                  #   debug - compile with no optimization
+                  #   (empty) - compile regularly
+
+    if [[ ! -f "$FILE" ]]; then
+      echo "$FILE not found"
+      exit 1
+    fi
+
+    if [[ "$MODE" == "all" ]]; then
+      SRCS=$(find . -name '.ccls-cache' -type d -prune -o -type f -name '*.c' -print)
+    else
+      SRCS="$FILE"
+    fi
+  
+    if [[ "$DEBUG" == "debug" ]]; then
+      CFLAGS="$CFLAGS -O0"
+    fi
+
+    rm -f ''$FILE.bin
+
+    set -o xtrace
+    ${clang}/bin/clang $CFLAGS $SRCS -o "''$FILE.bin"
+  '';
+  dap-cpp = pkgs.callPackage ../pkgs/dap-cpp { };
+  dapInitializeMessage = {
+    command = "initialize";
+    type = "request";
+    arguments = {
+      adapterID = "cppdbg";
+      clientID = "replit";
+      clientName = "replit.com";
+      columnsStartAt1 = true;
+      linesStartAt1 = true;
+      locale = "en-us";
+      pathFormat = "path";
+      supportsInvalidatedEvent = true;
+      supportsProgressReporting = true;
+      supportsRunInTerminalRequest = true;
+      supportsVariablePaging = true;
+      supportsVariableType = true;
+    };
+  };
+  dapLaunchMessage = program: {
+    command = "launch";
+    type = "request";
+    arguments = {
+      MIMode = "gdb";
+      arg = [ ];
+      cwd = ".";
+      environment = [ ];
+      externalConsole = false;
+      logging = { };
+      miDebuggerPath = "gdb";
+      name = "gcc - Build and debug active file";
+      inherit program;
+      request = "launch";
+      setupCommands = [
+        {
+          description = "Enable pretty-printing for gdb";
+          ignoreFailures = true;
+          text = "-enable-pretty-printing";
+        }
+      ];
+      stopAtEntry = false;
+      type = "cppdbg";
+    };
+  };
+in
+{
+  name = "C Tools";
+  version = "1.0";
+
+  packages = [
+    clang
+  ];
+
+  replit.runners.clang-project = {
+    name = "Clang: Project";
+    compile = "${compile}/bin/compile main.c all";
+    fileParam = false;
+    language = "c";
+    start = "./main.c.bin";
+  };
+
+  replit.runners.clang-single = {
+    name = "Clang: Single File";
+    compile = "${compile}/bin/compile $file single";
+    fileParam = true;
+    language = "c";
+    start = "./\${file}.bin";
+  };
```

Comment:
@ryantm made a good point: right now we don't support multiple runners. Maybe comment one of these out and make a TODO?

(i wanted to do the same for ruby)

Reply:
You mean it's not opened up to all users? But it'll be available to users with the flag. And it'll still work for users without the flag, they'll just use the top runner config.

Reply:
> they'll just use the top runner config

Nix attrsets are based on alphabetical ordering (ie, if you define `clang-single` *then* `clang-project` you'll still get `clang-project`). I don't think it's good to rely on this behavior - what if we want to add another runner? Then we'd have to validate that the previous behavior is alphabetically "lower" than whatever runner we're adding.

Reply:
Sorting is done in the frontend. But it's true that the current ordering is not the best for the users that don't have multi-runners, because the single file runner can go before the project runner depending if a file is open...

Maybe we need to have priority scheme for the runners...

Okay. I'll remove the single runners and defer that.